### DISCUSSION TITLE
Option picker - Edit/save row

### DISCUSSION THREAD

**Discussion by TobiasBorge** [Budibase/budibase#11403] [Help] [answered]
How do i get my option picker to update information in the relevant datatable?
I have connected the option picker to my answers column in my relevant table, but the answers (yes/no) will not update themselves when I try to click on them in preview. How do i solve this problem?

**Comment by ConorWebb96** [comment]
Hey @TobiasBorge,

I don't completely understand what you are asking here. Could you send some screenshots and further explain your issue?

Thanks!

**Reply by TobiasBorge** [reply]
![picker](https://github.com/Budibase/budibase/assets/140819491/24c1f817-9920-43ca-a119-138097d4fa05)

We have an option picker where the user should be able to answer questions from a table called "questions" with a yes/no answer. 
The problem is that when we click yes/no, the table cell containing the question from the relevant row in the "answer" column is not updated/saved. Do we maybe need to set a relationship between the option picker and the correct table/column/row, and if so, how do we do this?
Best regards, 
TobiasBorge

**Comment by TobiasBorge** [comment]
![picker2](https://github.com/Budibase/budibase/assets/140819491/2607cf4f-96cb-409b-b508-e62309731d73)

Here is our whole screen

**Comment by TobiasBorge** [comment]
![image](https://github.com/Budibase/budibase/assets/140819491/9d35e779-8aba-4578-96db-4952ba18aaf4)

To add content: When we try to answer the questions in "preview", one new row is added to our table with the answer no regardless of us clicking yes or no. 
We do not want rows to be added, just that rows are updated.

**Reply by ConorWebb96** [reply]
Hey @TobiasBorge,

I've spent a bit of time looking at this. The reason why this is likely not working is because you can't dynamically generate form fields within Budibase. I was looking at ways you could possibly do this from the information you've provided. The issue on this can be found here; https://github.com/Budibase/budibase/issues/5658. 

I was trying to create an array in a state value that could be continuously added to, it would have all the information needed to create a linked row between the questions and answers. Then i would send this back to an automation via an automation trigger action, it would run through the automation steps to create the row. However, I realised after a while that if I selected no for all questions it wouldn't update the array this is likely down to it seeing the field as the same field and as no change to the value happened nothing came through.

The only way i got this to work was to take everything out of the repeater and bind everything up manually. This only works if you have a fixed number of questions. 

I hope this explanation at least helps.

**Reply by TobiasBorge** [reply]
Thank you for taking the time @ConorWebb96 ,

That helped a lot! We did not understand how to bind everything up manually, but we managed to set conditions so that the data providers would only access one specific row each. 
We now manage to change the values in the data sheet, but it seems like each answer is still connected to each other inside the form somehow.

**Reply by ConorWebb96** [reply, marked as answer]
Morning @TobiasBorge,

I figured out a way to do this dynamically. I'll link the recipe export below; have a look through the export and feel free to ask any questions if needed.

[Questionaire-export-1690958840334.tar.gz](https://github.com/Budibase/budibase/files/12236923/Questionaire-export-1690958840334.tar.gz)

I hope this helps you in some way!

**Reply by TobiasBorge** [reply]
Thank you, @ConorWebb96!
I will take a look:)
I also figured out a way based on your previous answer where I use a data provider --> repeater --> form --> option picker.
I of course need to sort and set conditions for every question, but at least it worked.

**Reply by ConorWebb96** [reply]
Happy to hear it @TobiasBorge. Glad to have helped somewhat! 🚀
